components/brave_rewards/browser/database/database_contribution_queue.cc

```diff
@@ -49,6 +49,14 @@ bool DatabaseContributionQueue::Init(sql::Database* db) {
     return false;
   }
 
+  // We need to clear queue if user has corrupted foreign key
+  // more info: https://github.com/brave/brave-browser/issues/7579
+  if (publishers_->HasCorruptedForeignKey(db)) {
+    if (!DeleteAllRecords(db)) {
+      return false;
```

Comment:
`NOTREACHED()` as this should never happen?

Reply:
this can happen if something is wrong with db or locked